README.md

```diff
@@ -1,4 +1,52 @@
-# README
-To run the application, 
-run "./bin/dev" and in the terminal they mention the port link, 
-use that link to run the application on localhost
\ No newline at end of file
+# Newsletter Application
+ 
+## Overview
+
+This is a full-stack web application for posting articles. Built with [Ruby on Rails](https://rubyonrails.org/), [React](https://reactjs.org/), [PostgreSQL](https://www.postgresql.org/) and [Patternfly](https://www.patternfly.org/v4/).
+
+## Prerequisites
+
+Before the installation, please make sure to have global installations of-
+
+- node v16 or higher
+- npm v9 or higher
+- PostgreSQL (if running a local DB instance)
+- Ruby On Rails (ruby@3.1.2 & rails@7.0.4)
+
+## Set up development environment
+
+1. Get the code. Clone this git repository:
+
+    ```
+    git clone https://github.com/girijaasoni/newsletterApplicaton.git
+    cd newsletterApplicaton
+    ```
+
+2. Execute `bundle install && npm install` to configure the local environment.
+3. Update the DB configuration in `config/database.yml`
+4. Perform DB initialization/migration and seeding
+
+    ```
+    $ rails db:create
+    $ rails db:migrate
+    $ rails db:seed
+    ```
+
+5. Start the development server `rails s`
```

Comment:
I think in our case we need to run `"./bin/dev"`,
this can be added in another PR, cc @girijaasoni